packages/@sanity/form-builder/src/inputs/files/common/ActionsMenu.tsx

```diff
@@ -11,8 +11,8 @@ interface Props {
   onReset: MouseEventHandler<HTMLDivElement>
   accept: string
   directUploads: boolean
-  downloadUrl: string
-  copyUrl: string
+  downloadUrl?: string
+  copyUrl?: string
```

Comment:
I understand this will menu now is only being used in FileInputs (which is a different question that I brought up in the ImageInput). In what situations will we not have the downloadUrl / copyUrl for files?

Reply:
It is still being used for both File and Image inputs. They can be undefined when called in `renderAssetMenu()` - although that doesn't seem to be called if this is an invalid image.

I mimicked the v3 change here - not sure why it is like that though.

Reply:
Ahh you're right, I just double checked and forgot that there are two "menu" type things in the image input. I resolve the other comment (unfortunate naming on my part for these things, whoops)

As for the optional props, I'd like to understand why (maybe if @bjoerge or @rexxars have the time) but it isn't a blocker for this :)

Reply:
Ah, I believe the reason it is like this is that the v3 version has different setups for rendering preview vs field vs input, so this extra check makes sense. I'll change these to not be optional, since they are always set in v2.